### PR TITLE
KAFKA-2790: doc improvements

### PR DESCRIPTION
**Pull request by gwenshap**

### CODE REVIEW

File: docs/security.html [hunk 1]
```diff
@@ -261,6 +261,139 @@
...
+        <td>Resource</td>
+    </tr>
+    <tr>
+        <td>--consumer-group [group-name]</td>
+        <td>Specifies the consumer-group as resource.</td>
+        <td></td>
+        <td>Resource</td>
+    </tr>
+    <tr>
+        <td>--allow-principals</td>
```
Comment (junrao) [inline docs/security.html]: In KAFKA-2788, to support the SSL users, we will only allow a single principal to be specified through the allow and the deny options. So, we need to change the command line options accordingly. We will also need to change the examples accordingly (one command per user).

File: docs/security.html [hunk 2]
```diff
@@ -261,6 +261,139 @@
...
+        <pre>bin/kafka-acls.sh --authorizer kafka.security.auth.SimpleAclAuthorizer --authorizer-properties zookeeper.connect=localhost:2181 --add --allow-principals User:* --allow-hosts * --deny-principals User:BadBob --deny-hosts bad-host --operations Read--topic Test-topic</pre>
+        Above examples add acls to a topic by specifying --topic [topic-name] as the resource option. Similarly user can add acls to cluster by specifying --cluster and to a consumer group by specifying --consumer-group [group-name].</li>
+
+    <li><b>Removing Acls</b><br>
+            Removing acls is pretty much same , the only difference is instead of --add option users will have to specify --remove option. To remove the acls added by the first example above we can execute the CLI with following options:
```
Comment (junrao) [inline docs/security.html]: Minor edits: Removing acls is pretty much [the] same. [T]he only

File: docs/security.html [hunk 3]
```diff
@@ -261,6 +261,139 @@
 </ol>
 
 <h3><a id="security_authz">7.4 Authorization and ACLs</a></h3>
+Kafka ships with a pluggable Authorizer and an out-of-box authorizer implementation that uses zookeeper to store all the acls. Kafka acls are defined in the general format of "Principal P is [Allowed/Denied] Operation O From Host H On Resource R". You can read more about the acl structure on KIP-11. In order to add, remove or list acls you can use the Kafka authorizer CLI.
```
Comment (junrao) [inline docs/security.html:264]: Perhaps it's useful to add a link to KIP-11.

**Comment by guozhangwang** [comment]
Some general comments:

1) We could also change all the hrefs to the 0.9.0 java docs in api.html (e.g. new producer section).
2) We could remove the migration section as a whole since it is from xx -> 0.8, or change the section as "if your current version is older than 0.8.0, before upgrade to 0.9.0 you need to do a one-time migration following instructions here {href}" in the upgrade.html.

File: docs/security.html (hunk 3 above)
Comment (gwenshap) [inline docs/security.html:264]: I prefer not to mix user-facing and design docs. 

If there is information in KIP-11 that we believe users will need, we should add it here.

**Comment by gwenshap** [comment]
@junrao and @guozhangwang - addressed your suggestions.

**Comment by guozhangwang** [comment]
Also the auto generated data table seems ill-formated in browser (or is it only my browser)? http://kafka.apache.org/090/documentation.html#producerapi

**Comment by junrao** [comment]
Thanks for the patch. LGTM

File: docs/quickstart.html [hunk 4]
```diff
@@ -21,11 +21,11 @@
 
 <h4> Step 1: Download the code </h4>
 
-<a href="https://www.apache.org/dyn/closer.cgi?path=/kafka/0.8.2.0/kafka_2.10-0.8.2.0.tgz" title="Kafka downloads">Download</a> the 0.8.2.0 release and un-tar it.
+<a href="https://www.apache.org/dyn/closer.cgi?path=/kafka/0.9.0.0/kafka_2.10-0.9.0.0.tgz" title="Kafka downloads">Download</a> the 0.9.0.0 release and un-tar it.
```
Comment (ijuma) [inline docs/quickstart.html:24]: Should we be using kafka_2.11-0.9.0.0?

**Pull request closed without merging**